piccolo/query/base.py

```diff
@@ -87,6 +87,13 @@ def _validate(self):
         """
         pass
 
+    def __await__(self):
+        """
+        If the user doesn't explicity call .run(), proxy to it as a
+        convenience.
+        """
+        return self.run().__await__()
```

Comment:
if you want to keep respecting `run`'s `in_pool` flag in `__await__`, you can add `await_in_pool` property flag (setter+getter) to `Query`

Reply:
@unwinx How do you envisage that working?

Like this, or something else?

```python
query = MyTable.select()
query.in_pool = False
await query
```

Reply:
yes, exactly. as a "just in case" feature

Reply:
I'll leave this out for now - people can use `.run()` if they need more control.